lib/execution_engine2/SDKMethodRunner.py

```diff
@@ -312,16 +322,26 @@ def add_job_logs(self, job_id, log_lines):
         :return:
         """
         logging.debug(f"About to add logs for {job_id}")
-        job = self.get_mongo_util().get_job(job_id=job_id)
-        self._test_job_permissions(job, job_id, JobPermissions.WRITE)
+        mongo_util = self.get_mongo_util()
+
+        if not self._get_job_permission_from_cache(job_id, self.user_id, JobPermissions.WRITE):
+            with mongo_util.mongo_engine_connection():
+                job = Job.objects.with_id(job_id)
+                if not job:
+                    raise RecordNotFoundException(
+                        "Cannot find job log with id: {}".format(job_id)
+                    )
+            self._test_job_permissions(job, job_id, JobPermissions.WRITE)
+
         logging.debug("Success, you have permission to add logs for " + job_id)
 
-        try:
-            log = self.get_mongo_util().get_job_log(job_id=job_id)
-        except RecordNotFoundException:
-            log = self._create_new_log(pk=job_id)
+        with mongo_util.me_collection(self.config["mongo-logs-collection"]):
+            try:
```

Comment:
can we move this with into the get_job_log_pymongo function?

Reply:
👍